Fix HTTPS iframe and embed

xipasduarte opened the pull request:
Corrects the name of the methods that were being called.
- oembed_result -> secure_oembed_result
- embed_oembed_html -> secure_embed_oembed_html

Version: major version increment to 2.0.0, because of breaking changes.

xipasduarte committed:
Fix HTTPS iframe and embed

Corrects the name of the methods that were being called.

- oembed_result -> secure_oembed_result
- embed_oembed_html -> secure_embed_oembed_html

Version: major version increment to 2.0.0, because of breaking changes.

s3rgiosan commented on wp-content/mu-plugins/https.php:
```diff
@@ -2,7 +2,7 @@
  * Plugin Name: HTTPS
  * Description: Fixes and improves HTTPS support.
- * Version:     1.0.0
+ * Version:     2.0.0
```
Well, I don't think that this is a change for an increment of the major version. It's more like a minor version or a patch version. But definitely not a major version.

s3rgiosan replied:
And I don't think this is a breaking change because it's not used anywhere else except on the defined filters on the bottom

s3rgiosan commented on wp-content/mu-plugins/https.php:
```diff
@@ -25,25 +26,27 @@ function wp_calculate_image_srcset( $sources ) {
  * Replace http:// with https:// in the embed code (before caching).
- * 
+ *
+ * @since  2.0.0
```
And this should be something like this:

```
@since  1.X.X Changed the name of the method for something more meaningfully. 
@since  1.0.0
```

xipasduarte replied:
Was going to ask about that, I thought it must have been something like that, but I got lazy 👅 .

As for the version, if you consider it in the right perspective you will find that the major is actually required. The names of the functions changed, and you are saying that they're only used here, but the fact is the "API" changed and it would break the usage of this plugin anywhere else.
I know it's a part of this repository, but one must not discard the chance of someone copy pasting this for their projects (😂  eheh). After all this is an isolated plugin.

I considered changing the calls in the `add_action`s, in this case it would be a minor, but I think the names are more in line with the plugin's purpose. Still, there is time 😃 .

s3rgiosan replied:
@xipasduarte I understand your arguments but I have mixed feelings about this. 

I think we should adopt the WooCrap way... badum tschhh 😝 

Let's see what the "others" have to say. Just remember that we use the [Semantic Versioning](http://semver.org/).

xipasduarte replied:
I was on a "minor" road, then I remembered that and went to the site which put me on a "major" eheh. It's just because of the "API" thing, but concede that it's a bit far fetched in a way 😄 .

Let the other speak then... others?

xipasduarte committed:
Correct doc_block

Correct previous documentation block changes. Adds all version for a method with details on the changes.
